darshan-runtime/lib/darshan-hdf5.c

```diff
@@ -364,6 +364,9 @@ herr_t DARSHAN_DECL(H5Fflush)(hid_t object_id, H5F_scope_t scope)
                     tm1, tm2, rec_ref->last_meta_end);
             }
             H5F_POST_RECORD();
+
+            MAP_OR_FAIL(H5Fclose);
```

Comment:
Should we move this inside the `if` block above just to be safe? If `H5Iget_file_id` returns an error (i.e., `H5I_INVALID_HID`), that could lead to some noisy warnings/errors passing that value in to `HF5close`.

Reply:
Isn't that already inside of `if(file_id > 0)` block?

Reply:
Oh wow, you're right. Not sure what I was thinking, probably just moving too fast. Thanks for the correction.